### PR TITLE
[TASK] Revise Caching/Developer chapter

### PR DESCRIPTION
**Pull request by brotkrueml**
- Adjust line length
- Add captions to code snippets
- Add default configuration to Services.yaml snippet as people tend to copy/paste that part and without it it does not work
- Add note about flushing the cache

Releases: main, 11.5

### CODE REVIEW

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst [hunk 1]
```diff
@@ -1,143 +1,165 @@
...
-for their needs. It is about how to use the framework properly. For details about
-its inner working, please refer to the :ref:`section about architecture <caching-architecture>`.
+This chapter is targeted at extension authors who want to use the caching
+framework for their needs. It is about how to use the framework properly. For
+details about its inner working, please refer to the
+:ref:`section about architecture <caching-architecture>`.
 
-Example usages can be found throughout the TYPO3 CMS Core, in particular in
-system extension `core` and `extbase`.
+Example usages can be found throughout the TYPO3 Core, in particular in system
```
Comment (linawolf) [inline Documentation/ApiOverview/CachingFramework/Developer/Index.rst:14]:
```suggestion
Example usages can be found throughout the TYPO3 Core, in particular in the system
```

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst [hunk 2]
```diff
@@ -1,143 +1,165 @@
...
 Cache Registration
 ==================
 
-Registration of a new cache should be done in :file:`ext_localconf.php`. The example below just defines
-an empty sub-array in *cacheConfigurations*. Neither *frontend* nor *backend* are defined,
-meaning that the cache manager will choose the default :ref:`variable frontend <caching-frontend-variable>`
-and the :ref:`database backend <caching-backend-db>` by default.
-
-.. code-block:: php
+Registration of a new cache should be done in :file:`ext_localconf.php`. The
```
Comment (linawolf) [inline Documentation/ApiOverview/CachingFramework/Developer/Index.rst:24]:
```suggestion
Registration of a new cache should be done in an extension's :ref:`ext-localconf-php`. The
```

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst [hunk 3]
```diff
@@ -1,143 +1,165 @@
...
+example below just defines an empty sub-array in *cacheConfigurations*. Neither
```
Comment (linawolf) [inline Documentation/ApiOverview/CachingFramework/Developer/Index.rst:25]:
```suggestion
example below defines an empty sub-array in `cacheConfigurations`. Neither
```

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst [hunk 4]
```diff
@@ -1,143 +1,165 @@
...
+*frontend* nor *backend* are defined, meaning that the cache manager will choose
```
Comment (linawolf) [inline Documentation/ApiOverview/CachingFramework/Developer/Index.rst:26]:
```suggestion
*frontend* nor *backend* are defined: The cache manager will choose
```

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst [hunk 5]
```diff
@@ -1,143 +1,165 @@
...
+the default :ref:`variable frontend <caching-frontend-variable>` and the
+:ref:`database backend <caching-backend-db>` by default.
 
-   $GLOBALS['TYPO3_CONF_VARS']['SYS']['caching']['cacheConfigurations']['myext_mycache'] ??= [];
+..  code-block:: php
+    :caption: EXT:some_extension/ext_localconf.php
```
Comment (linawolf) [inline Documentation/ApiOverview/CachingFramework/Developer/Index.rst:31]:
```suggestion
    :caption: EXT:my_extension/ext_localconf.php
```
Should the extension and the array key be alike? EXT:my_extension -> myext_mycache?

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst [hunk 6]
```diff
@@ -1,143 +1,165 @@
...
 
-   The null coalescing assignment operator (:code:`??=`) check is used to enable administrators to overwrite configuration of caches in
-   :file:`config/system/settings.php`. During bootstrap, any :file:`ext_localconf.php` is loaded **after**
-   :file:`DefaultConfiguration.php` and :file:`config/system/additional.php` are loaded, so it is
-   important to make sure that the administrator did not already set any configuration of the
-   extensions cache.
+..  tip::
+    The null coalescing assignment operator (:code:`??=`) check is used to
+    enable administrators to overwrite configuration of caches in
+    :file:`config/system/settings.php`. During :ref:`bootstrap <bootstrapping>`,
```
Comment (linawolf) [inline Documentation/ApiOverview/CachingFramework/Developer/Index.rst:39]:
```suggestion
    :ref:`config/system/settings.php <configuration-files>`. During :ref:`bootstrap <bootstrapping>`,
```

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst [hunk 7]
```diff
@@ -1,143 +1,165 @@
...
+:yaml:`cache.myext_mycache`, the name of the cache is `myext_mycache` (as
+defined in :php:`ext_localconf.php`). Both can be anything you like, just make
+sure they are unique and clearly hint at the purpose of your cache.
 
-.. note::
-
-   Since TYPO3v10, you should prefer :ref:`dependency injection <DependencyInjection>`
-   to retrieve cache frontends whenever possible and no longer use the :php:`CacheManager` directly.
+..  note::
+    Since TYPO3v10, you should prefer :ref:`dependency injection
```
Comment (linawolf) [inline Documentation/ApiOverview/CachingFramework/Developer/Index.rst:100]: I would remove the reference to version 10 as we are in 12 here and 9 is already not supported anymore

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst [hunk 8]
```diff
@@ -1,143 +1,165 @@
...
+    <DependencyInjection>` to retrieve cache frontends whenever possible and no
+    longer use the :php:`CacheManager` directly.
 
 Here is some example code which retrieves the cache via dependency injection:
 
+..  code-block:: php
+    :caption: EXT:some_extension/Classes/MyClass.php
```
Comment (linawolf) [inline Documentation/ApiOverview/CachingFramework/Developer/Index.rst:107]: I would put this code into a file

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst [hunk 9]
```diff
@@ -1,143 +1,165 @@
...
 
-.. tip::
+..  tip::
+    It is not needed to call :php:`$this->cache->has()` before accessing cache
+    entries with :php:`$this->cache->get()` as the latter returns :php:`false`
+    if no entry exists.
 
-   It isn't needed to call :code:`has()` before accessing cache entries with :code:`get()`
-   as the latter returns :code:`false` if no entry exists.
+Since the auto-wiring feature of the dependency injection container cannot
```
Comment (linawolf) [inline Documentation/ApiOverview/CachingFramework/Developer/Index.rst:144]: Maybe link to the according section in the DI chapter?

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst [hunk 10]
```diff
@@ -1,143 +1,165 @@
...
 
-Here `@cache.myext_mycache` refers to the cache service we defined above. This setup allows you
-to freely inject the very same cache into any class.
+Here :yaml:`@cache.myext_mycache` refers to the cache service we defined above.
+This setup allows you to freely inject the very same cache into any class.
+
+..  note::
+    After changes in the :file:`Services.yaml` file flush the cache via
+    :guilabel:`Admin Tools > Maintenance` or the :ref:`CLI command
+    <symfony-console-commands>` `cache:flush`.
```
Comment (linawolf) [inline Documentation/ApiOverview/CachingFramework/Developer/Index.rst:165]: Put the full command here as convenience to copy as code-block? At least I love doing that ;)

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst (hunk 5 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/CachingFramework/Developer/Index.rst:31]: No, below is some_extension already used. So I chose to use it also here.

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst (hunk 7 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/CachingFramework/Developer/Index.rst:100]: PR is merged into v11. I will revise it in a next step only for v12 (constructor property promotion) and want to remove it then.

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst (hunk 8 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/CachingFramework/Developer/Index.rst:107]: In the next step with constructor property promotion. This is just a clean-up before adjusting for v12.

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst (hunk 5 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/CachingFramework/Developer/Index.rst:31]: Done

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst (hunk 9 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/CachingFramework/Developer/Index.rst]: Done

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst (hunk 10 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/CachingFramework/Developer/Index.rst:166]: Hm, then we have to add also the legacy path with tabs which makes it a little bit big just for clearing the cache.

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst (hunk 8 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/CachingFramework/Developer/Index.rst]: Aah, know why I didn't do it. It is not valid PHP:

```php
$value = /* ... Logic to calculate value ... */;
```

Thought about moving it into methods but that would be overkill.

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst (hunk 10 above)
Comment (linawolf) [reply Documentation/ApiOverview/CachingFramework/Developer/Index.rst:166]: We should just make a template for clearing the cache once and then include it wherever needed?

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst (hunk 8 above)
Comment (linawolf) [reply Documentation/ApiOverview/CachingFramework/Developer/Index.rst]: So can we give an example value instead?

File: Documentation/ApiOverview/CachingFramework/Developer/Index.rst (hunk 10 above)
Comment (linawolf) [reply Documentation/ApiOverview/CachingFramework/Developer/Index.rst:166]: can be done in a follow up though

**Review by linawolf — APPROVED** [review]

**Pull request merged**